### PR TITLE
Add simple feature toggles table and GET endpoint

### PR DESCRIPTION
**Pull request by Adam-C-Mitchell**
Addition of a feature toggles table and the ability to retrieve them from the database by name. This will be used for any feature toggles needed in future, but the primary instigator behind this was the need for an informative banner that tells users that the service is not currently live, that can be switched off/changed any time from the database.

### CODE REVIEW

**Review by jonathan-owens-gds — APPROVED** [review]

**Review by jonathan-owens-gds — CHANGES_REQUESTED** [review]
TLDR: Could we rename the toggle to be "show-service-mode-banner"? The current name of the toggle makes it seem as if it's controlling whether the service is live or not. Otherwise: Good implementation - the feature toggle will be useful going forward.

---
I'm not sure the feature toggle should be how we determine whether the service is live or not. There are the 'restricted' and 'go_live_at' fields in the 'services' table that should indicate whether the service mode is "live" or "training".
If the service isn't live, someone could still change this toggle to say it is live, which would be inconsistent.
I think the feature toggle should be something like "show-service-mode-banner" - if the banner is the "feature" then the toggle should just toggle this on or off. The mode of the service, live or training, isn't a feature, rather a status of the service and shouldn't be toggled but managed through the platform admin's UI.

**Comment by jonathan-owens-gds** [comment]
So this banner would be shown on the 'preview' and 'staging' environments, but not on 'prod'? I guess if Katia and Mehmet have signed off on this feature then that's fine - I don't understand the use-case/need for it, if it isn't linked to the service's modes (e.g., live, training).

**Review by jonathan-owens-gds — APPROVED** [review]

**Comment by jonathan-owens-gds** [comment]
In the feature toggle object, the final column is "display_html" - but I only see text in there.  Is there a use-case where this will be used to inject actual html into the information banner?
If not, could it be renamed to "display_text"?
If it can be limited to text only, that would be safer. Otherwise any html could be inserted by this field, which may reduce our control over the styling and layout.

**Comment by Adam-C-Mitchell** [comment]
> If Katia and Mehmet have signed off on this feature then that's fine - I'm not sure I understand what this is for, if it isn't linked to the service's modes (e.g., live, training).

It's basically to make sure people know that the application hasn't been enabled yet in production - There's a window where the application is rolled out (for training purposes etc.) but without switching on the actual alerting functionality, and we don't want people trying to send alerts out and it not doing anything.

> In the feature toggle object, the final column is "display_html" - but I only see text in there. Is there a use-case where this will be used to inject actual html into the information banner? If not, could it be renamed to "display_text"? If it can be limited to text only, that would be safer. Otherwise any html could be inserted by this field, which may reduce our control over the styling and layout.

Well, I don't think there would be a use case for that in the informative banner (the design system component that the text is displayed in takes care of the need for HTML), but it was a suggestion that feature toggles in future might. I would agree (I mean, I'm not concerned, because the HTML can only be changed directly from the database, but still), text would be better. I think for now it might be wiser to have the field as "display_text" as you say and if in future we do find that there is a use case for HTML, we can just change it then - I don't think that would be too much trouble to do. What do you think?

**Comment by jonathan-owens-gds** [comment]
If there was discussion of a possible future requirement, then I'm fine if we leave it as "display_html".

**Review by Adam-C-Mitchell — COMMENTED** [review]
Changing migration to set service_is_not_live to false because it should be getting switched on as and when is needed via the database, rather than automatically being switched on
